package/MDAnalysis/lib/picklable_file_io.py

```diff
@@ -0,0 +1,535 @@
+    -------
+    ::
+
+        >>> file = BZ2Picklable(XYZ_bz2)
+        >>> file.readline()
+        >>> file_pickled = pickle.loads(pickle.dumps(file))
+        >>> print(file.tell(), file_pickled.tell())
+            5 5
+
+    See Also
+    ---------
+    FileIOPicklable
+    BufferIOPicklable
+    TextIOPicklable
+    GzipPicklable
+
+
+    .. versionadded:: 2.0.0
+    """
+    def __init__(self, name, mode='rb'):
+        self._bz_mode = mode
+        super().__init__(name, mode)
+
+    def __getstate__(self):
+        if not self._bz_mode.startswith('r'):
+            raise RuntimeError("Can only pickle files that were opened "
+                               "in read mode, not {}".format(self._bz_mode))
+        return self._fp.name, self.tell()
+
+    def __setstate__(self, args):
+        super().__init__(args[0])
+        self.seek(args[1])
+
+
+class GzipPicklable(gzip.GzipFile):
+    """Gzip file object (read-only) that can be pickled.
+
+    This class provides a file-like object (as returned by :func:`gzip.open`,
+    namely :class:`gzip.GzipFile`) that, unlike standard Python file objects,
+    can be pickled. Only read mode is supported.
+
+    When the file is pickled, filename and position of the open file handle in
+    the file are saved. On unpickling, the file is opened by filename,
+    and the file is seeked to the saved position.
+    This means that for a successful unpickle, the original file still has to
+    be accessible with its filename.
+
+    Note
+    ----
+    This class only supports reading files in binary mode. If you need to open
+    to open a compressed file in text mode, use the :func:`gzip_pickle_open`.
+
+    Parameters
+    ----------
+    name : str
+        a filename given a text or byte string.
+    mode : str
+        can only be 'r', 'rb' to make pickle work.
+
+    Example
+    -------
+    ::
+
+        >>> file = GzipPicklable(MMTF_gz)
+        >>> file.readline()
+        >>> file_pickled = pickle.loads(pickle.dumps(file))
+        >>> print(file.tell(), file_pickled.tell())
+            1218 1218
+
+    See Also
+    ---------
+    FileIOPicklable
+    BufferIOPicklable
+    TextIOPicklable
+    BZ2Picklable
+
+
+    .. versionadded:: 2.0.0
+    """
+    def __init__(self, name, mode='rb'):
+        self._gz_mode = mode
+        super().__init__(name, mode)
+
+    def __getstate__(self):
+        if not self._gz_mode.startswith('r'):
+            raise RuntimeError("Can only pickle files that were opened "
+                               "in read mode, not {}".format(self._gz_mode))
+        return self.name, self.tell()
+
+    def __setstate__(self, args):
+        super().__init__(args[0])
+        self.seek(args[1])
+
+
+def pickle_open(name, mode='rt'):
+    """Open file and return a stream with pickle function implemented.
+
+    This function returns either BufferIOPicklable or TextIOPicklable wrapped
+    FileIOPicklable object given different reading mode. It can be used as a
```

Comment:
```suggestion
    This function returns a FileIOPicklable object wrapped in a BufferIOPicklable
    class when given the "rb" reading mode, or a FileIOPicklable object wrapped in
    a TextIOPicklable class with the "r" reading mode. It can be used as a
```
👆 at least that's my understanding of it, but I did have to think a bit to work that out. It's always better for docs to be as explicit as possible.